### PR TITLE
Disable interaction with players while Dragon is alive.

### PR DESCRIPTION
**Pull request by ShiroSR**
Opened an issue asking for this a few hours ago, ended up doing it myself with my non-existent knowledge of Java and coding in general.
Didn't hide the players because it's counterproductive.

### CODE REVIEW

**Comment by biscuut** [comment]
Hi thanks for this, but maybe this is too specific? Can I change it maybe to disallow clicking players when you are holding out a bow?

File: src/main/java/codes/biscuit/skyblockaddons/mixins/MixinEntityRenderer.java [hunk 1]
```diff
@@ -49,7 +49,7 @@ private void removeEntities(List<Entity> list) {
             if (!SkyblockAddons.getInstance().getConfigValues().isDisabled(Feature.DISABLE_PROFILE_VIEW_IF_DRAGON_ALIVE)) {
                 Minecraft mc = Minecraft.getMinecraft();
                 for (Entity elist : mc.theWorld.loadedEntityList) {
-                    if (elist instanceof EntityEnderman) {
+                    if (elist instanceof EntityDragon) {
                         double coreX = -670;
```
Comment (ViralTaco) [inline src/main/java/codes/biscuit/skyblockaddons/mixins/MixinEntityRenderer.java]: Shouldn't this be final?

File: src/main/java/codes/biscuit/skyblockaddons/mixins/MixinEntityRenderer.java [hunk 2]
```diff
@@ -43,6 +46,18 @@ private void removeEntities(List<Entity> list) {
...
             }
+            if (!SkyblockAddons.getInstance().getConfigValues().isDisabled(Feature.DISABLE_PROFILE_VIEW_IF_DRAGON_ALIVE)) {
+                Minecraft mc = Minecraft.getMinecraft();
+                for (Entity elist : mc.theWorld.loadedEntityList) {
+                    if (elist instanceof EntityDragon) {
+                        double coreX = -670;
+                        double coreY = 9;
+                        double coreZ = -275;
+
+                        list.removeIf(listEntity -> listEntity.getDistance(coreX, coreY, coreZ) <= 70 && listEntity instanceof EntityOtherPlayerMP);
```
Comment (ViralTaco) [inline src/main/java/codes/biscuit/skyblockaddons/mixins/MixinEntityRenderer.java]: Would checking if `listEntity instanceof EntityOtherPlayerMP` first be faster?

**Comment by ShiroSR** [comment]
> Hi thanks for this, but maybe this is too specific? Can I change it maybe to disallow clicking players when you are holding out a bow?

Way better idea than mine, didn't think about it.

File: src/main/java/codes/biscuit/skyblockaddons/mixins/MixinEntityRenderer.java (hunk 1 above)
Comment (ShiroSR) [reply src/main/java/codes/biscuit/skyblockaddons/mixins/MixinEntityRenderer.java]: I was testing with Endermans (Zealots) and forgot to change it back to Ender Dragon, the doubles should be fine though.

File: src/main/java/codes/biscuit/skyblockaddons/mixins/MixinEntityRenderer.java (hunk 2 above)
Comment (ShiroSR) [reply src/main/java/codes/biscuit/skyblockaddons/mixins/MixinEntityRenderer.java]: Probably better, thank you.

**Pull request closed without merging**